src/components/PopupWithForm.js

```diff
@@ -18,17 +19,51 @@ export default class PopupWithForm extends Popup {
     return formValues;
   }
 
+  _checkFormValidity() {
+    const isFormValid = Array.from(this._inputList).every(
+      (input) => input.value.trim() !== ""
+    );
+    this._submitButton.disabled = !isFormValid;
+    if (isFormValid) {
+      this._submitButton.classList.remove("button_inactive");
+    } else {
+      this._submitButton.classList.add("button_inactive");
+    }
+  }
+
   _setEventListeners() {
-    this._popupForm.addEventListener("submit", (event) => {
+    this._popupForm.addEventListener("submit", async (event) => {
       event.preventDefault();
       const data = this._getInputValues();
-      this._handleFormSubmission(data);
-      this.reset();
+      this.renderLoading(true);
+
+      await this._handleFormSubmission(data);
+
+      this.close();
+      this.renderLoading(false);
     });
+
+    this._inputList.forEach((input) => {
+      input.addEventListener("input", () => {
+        this._checkFormValidity();
+      });
+    });
+  }
+
+  open() {
+    super.open();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.